apis/src/main/java/org/apache/rocketmq/apis/consumer/PullConsumer.java

```diff
@@ -0,0 +1,186 @@
+     *
+     * <p>Nothing occurs if the specified topic does not exist in subscription expressions of pull consumer.
+     *
+     * @param topic the topic to remove subscription.
+     * @return pull consumer instance.
+     */
+    PullConsumer unsubscribe(String topic) throws ClientException;
+
+    /**
+     * Get the collection of messageQueues currently assigned to current consumer.
+     * @return the collection of messageQueues currently assigned to current consumer
+     */
+    Collection<MessageQueue> assignments();
+
+    /**
+     * Pull messages for the topics specified by subscribe or assign api synchronously.
+     *
+     * <p> This method returns immediately if there are messages available.
+     * Otherwise, it will await the passed timeout. If the timeout expires, an empty map will be returned.
+     * An error occurs if you do not subscribe or assign messageQueues before pulling for data.
+     * @param maxMessageNum max message num when server returns.
+     * @return map of messageQueue to messageViews.
+     */
+    Map<MessageQueue, Collection<MessageView>> pull(int maxMessageNum) throws ClientException;
+
+    /**
+     * Pull messages for the topics specified by subscribe or assign api asynchronously.
+     *
+     * <p> This method returns immediately if there are messages available.
+     * Otherwise, it will await the passed timeout. If the timeout expires, an empty map will be returned.
+     * An error occurs if you do not subscribe or assign messageQueues before pulling for data.
+     * @param maxMessageNum max message num when server returns.
+     * @return future of messageViews.
+     */
+    CompletableFuture<Map<MessageQueue, Collection<MessageView>>> pullAsync(int maxMessageNum) throws ClientException;
+
+    /**
+     * Commit offsets synchronously which returned by the last pull api for all the subscribed topic and messageQueues.
+     *
+     * <p> This offsets meta maintained by kafka, which used for rebalance processing or next startup.
+     * If you store the offsets in your own storage, you may need call seek api before pull and no need to call commit.
+     */
+    void commit() throws ClientException;
+
+    /**
+     * Commit offsets asynchronously which returned by the last pull api for all the subscribed topic and messageQueues.
+     *
+     * <p> This offsets meta maintained by kafka, which used for rebalance processing or next startup.
+     * If you store the offsets in your own storage, you may need call seek api before pull and no need to call commit.
+     */
+    CompletableFuture<Void> commitAsync() throws ClientException;
+
+    /**
+     * Commit specified offset for the specified messageQueue synchronously.
+     * @param messageQueue the specified messageQueue to commit offset
+     * @param committedOffset the specified offset commit to server
+     */
+    void commit(MessageQueue messageQueue, long committedOffset) throws ClientException;
```

Comment:
need commit  a collection of messageQueue  interface;
and commitedOffset  can not find in the return of pull method.